File: Tutorials/Parallelisation/code/to_order/mpi.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <string.h>
#include <time.h>
#include <mpi.h>

#define RANDMAX 10000

int main(int argc, char** argv) {

  int rank, my_val, nProc;
  int is_sorted, final_is_sorted;

  MPI_Init(&argc,&argv);
  MPI_Comm_rank(MPI_COMM_WORLD,&rank);

  srand(time(NULL)+100*rank);   // init random generator (adding rank to avoid same seed for multiple processes)

  my_val = rand() % RANDMAX; 

  // TODO exchange values of "neighboring" process(es) here

  // TODO compare values

  // TODO gather/reduce results here and store into final_is_sorted on rank 0
  
  if (rank == 0) {    
    printf("%d neighbouring sorted pairs found!\n",final_is_sorted);
  }

  MPI_Finalize();

}

File: Tutorials/Parallelisation/Code/Exercises/to_order/mpi.py
import mpi4py; mpi4py.rc.recv_mprobe = False
from mpi4py import MPI
import time
import random
import sys

RANDMAX = 10000

if __name__ == '__main__':

   comm = MPI.COMM_WORLD
   nProc = comm.Get_size()
   rank = comm.Get_rank()

   random.seed(time.time()+100*rank) # init random generator
   my_val = random.randint(0,RANDMAX-1)

   # TODO exchange values of "neighbouring" process(es) here

   # TODO compare values

   # TODO gather/reduce results here and store into final_is_sorted on rank 0

   if (rank == 0):
      print("%d neighbouring sorted pairs found!" % (final_is_sorted))




